### ISSUE TITLE
Handwriting support 

### ISSUE THREAD

**Issue by derei** [futo-org/android-keyboard#389]
### Use Cases

One of the most popular brands of mobile deviis Samsung. And Samsung does an exquisite job at providing stylus support, they partnered with Wacom, and they nailed it.
The only issue here is that if any Samsung device user wants to use a more private keyboard than Samsung Keyboard, must give up the handwriting support. 
By this, Samsung is practically holding users hostages.

And if that doesn't anger you enough:
 _Samsung is actively blocking tuta.com email accounts to be used to sign up for a Samsung account. And no, is not a filter to allow only "known providers". It is specifically targeted against Tuta. A fictional account such as "fdmghrskjfh@sektknfskjedn.com" will be accepted, but any tuta domain will be rejected._ 

Why did I bring tuta into this? Well just to enrage people, if that motivates some; so people know how Samsung plays. To give them more reasons to thwart Samsung monopoly. 

### Proposal

Implementing handwriting support competitive with Samsung Keyboard  and compatible with Samsung devices, would disrupt this monopoly and allow users to choose.

What makes Samsung Keyboard so unique, is that it allows both writing in place (in the actual text field), but it also provides a handwriting pad.

**Comment by The-Quantum-Alpha** [comment]
Even given my profession, i do believe this is highly complex to implement from scratch...
Nonetheless, i this this would be a great idea.

**Comment by SgtxRootbeer** [comment]
I just want to put my 2 cents in, this should not be limited to samsung devices. There are other devices, such as the google pixel tablet (or my fire max 11), which support styluses and their keyboards also support handwriting. Otherwise, this is a very nice addition.

**Comment by mikeloeven** [comment]
While I understand using Samsung as the most notable example of handwriting support, it's very narrow-minded and short-sighted to suggest that this feature only be added for Samsung phones. Any Android device with a capacitive touchscreen will support both passive and active styluses and will benefit from the addition of a handwriting input.

**Comment by derei** [comment]
@mikeloeven you are taking this the wrong way, which makes you the narrow minded one, I'm afraid. As Futo is not device-dependent, it would obviously work on any compatible device if a handwriting support system would be added.  However, Samsung Keyboard is device dependent and this is introducing a monopoly.  Hence the reason why Samsung was used as an example.

**Comment by mikeloeven** [comment]
Yeah I misinterpreted but honestly right now its either Samsung or Gboard if you dont have a samsung device and there might be some proprietary writing inputs AFAIK it does not seem to exist in anything open source so either way its needed and should not really be on the Low priority list

**Comment by derei** [comment]
@dean-futo oh hear thythe words of @mikeloeven, and fear not the complexity of handwritting implementation, for that you are brave and competent!

**Comment by Idcrafter** [comment]
Gboard also has hand writing and that also inside text boxes, it seems to also work offline on a Oneplus Pad 3 i had to test it with and i just disable networking via root for Gboard and it recognized what i wrote with the pen (in the keyboard ui for it and the Textfields in a app). 

Having small OCR system for it would make sense and having it retrain over time is also good but wouldn't it also be nicer to ask the user to first write most or least common letter (depending on what is smarter) for the Language they use to start adapting the ocr on device?
a ability to contribute hand written input (the least readable the better) from the community similar to the swipe typing contribution would be a way to gain a great data set for the beginning?
